src/main/java/de/craftlancer/clfeatures/jukebox/JukeboxSong.java

```diff
@@ -0,0 +1,179 @@
+package de.craftlancer.clfeatures.jukebox;
+
+import de.craftlancer.clfeatures.CLFeatures;
+import de.craftlancer.core.menu.ConditionalPagedMenu;
+import de.craftlancer.core.menu.MenuItem;
+import de.craftlancer.core.resourcepack.ResourcePackManager;
+import de.craftlancer.core.resourcepack.TranslateSpaceFont;
+import de.craftlancer.core.util.ItemBuilder;
+import de.craftlancer.core.util.Tuple;
+import org.bukkit.Material;
+import org.bukkit.configuration.serialization.ConfigurationSerializable;
+import org.bukkit.entity.Player;
+import org.bukkit.event.inventory.ClickType;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class JukeboxSong implements ConfigurationSerializable {
+    
+    // tick -> notes (every 20 ticks per second, 1 repeater tick = 2 MC ticks)
+    private Map<Integer, List<AbstractJukeboxNote>> notes = new HashMap<>();
+    private ConditionalPagedMenu tickOverviewMenu;
+    private UUID uuid;
+    
+    public JukeboxSong(UUID uuid) {
+        this.uuid = uuid;
+    }
+    
+    public JukeboxSong(Map<String, Object> map) {
+        
+        this.uuid = UUID.fromString((String) map.get("uuid"));
+        
+        map.remove("uuid");
+        map.remove("==");
```

Comment:
You shouldn't have to modify the deserialization map. Put the note data one layer deeper, maybe?